framework/src/base/Assembly.C

```diff
@@ -939,7 +943,12 @@ Assembly::prepare()
     unsigned int vi = ivar.number();
     unsigned int vj = jvar.number();
 
-    jacobianBlock(vi, vj).resize(ivar.dofIndices().size(), jvar.dofIndices().size());
+    AllLocalDofIndicesThread aldit(_sys.system(), {jvar.name()});
```

Comment:
Actually - it turns out that this _IS_ the problem.  We currently don't allow threaded loops to run within other threaded loops.

The mechanism that forbids this is here: https://github.com/libMesh/libmesh/blob/master/include/parallel/threads_pthread.h#L335

That is essentially a "lock" that doesn't allow two different threaded loops to execute.

You need to move this whole dof_indices finding thing somewhere more suitable... like maybe into the variable's themselves or into one of the `initialSetup()` routines in Assembly or FEProblem